llap-server/src/java/org/apache/hadoop/hive/llap/daemon/impl/LlapLoadGeneratorService.java

```diff
@@ -0,0 +1,98 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.hadoop.hive.llap.daemon.impl;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hive.conf.HiveConf;
+import org.apache.hadoop.service.AbstractService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.Random;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Extra load generator service for LLAP.
+ */
+public class LlapLoadGeneratorService extends AbstractService {
+  private static final Logger LOG = LoggerFactory.getLogger(LlapLoadGeneratorService.class);
+  private long interval;
+  private float threshold;
+  private String[] victimsHostName;
+  private Thread[] threads;
+
+  public LlapLoadGeneratorService() {
+    super("LlapLoadGeneratorService");
+  }
+
+  @Override
+  protected void serviceInit(Configuration conf) throws Exception {
+    super.serviceInit(conf);
+    threshold = HiveConf.getFloatVar(conf, HiveConf.ConfVars.HIVE_TEST_LOAD_UTILIZATION);
+    victimsHostName = HiveConf.getTrimmedStringsVar(conf, HiveConf.ConfVars.HIVE_TEST_LOAD_HOSTNAMES);
+    interval = HiveConf.getTimeVar(conf, HiveConf.ConfVars.HIVE_TEST_LOAD_INTERVAL, TimeUnit.MILLISECONDS);
+    LOG.info("LlapLoadGeneratorService init with {} {} {}", interval, threshold, victimsHostName);
+  }
+
+  @Override
+  protected void serviceStart() throws UnknownHostException {
+    String localHostName = InetAddress.getLocalHost().getHostName();
+    LOG.debug("Local hostname is: {}", localHostName);
+    for (String hostName : victimsHostName) {
+      if (hostName.equalsIgnoreCase(localHostName)) {
+        LOG.debug("Starting load generator process on: {}", localHostName);
+        threads = new Thread[Runtime.getRuntime().availableProcessors()];
+        Random random = new Random();
+        for (int i = 0; i < threads.length; i++) {
+          threads[i] = new Thread(new Runnable() {
+            @Override
+            public void run() {
+              while (!Thread.interrupted()) {
+                if (random.nextFloat() <= threshold) {
+                  // Keep it busy
+                  long startTime = System.currentTimeMillis();
+                  while (System.currentTimeMillis() - startTime < interval) {
```

Comment:
IMHO - it is not really important how we eat the CPU cycles, but checking the isInterrupted flag is really needed, so that is solved our other problem as well :)